infrastructure/main.tf

```diff
@@ -0,0 +1,145 @@
+  description = "Availability Zones"
+  default     = ["eu-central-1a", "eu-central-1b", "eu-central-1c"]
+}
+
+module "public_subnet_cider_blocks" {
+  source = "hashicorp/subnets/cidr"
+
+  base_cidr_block = var.vpc_cidr_block
+
+  networks = [
+    {
+      name     = "${local.subnet_prefix_name.private}_${var.azs[0]}"
+      new_bits = tonumber(local.subnet_cider_bits.private)
+    },
+    {
+      name     = "${local.subnet_prefix_name.private}_${var.azs[1]}"
+      new_bits = tonumber(local.subnet_cider_bits.private)
+    },
+    {
+      name     = "${local.subnet_prefix_name.private}_${var.azs[2]}"
+      new_bits = tonumber(local.subnet_cider_bits.private)
+    },
+    {
+      name     = "${local.subnet_prefix_name.db}_${var.azs[0]}"
+      new_bits = tonumber(local.subnet_cider_bits.db)
+    },
+    {
+      name     = "${local.subnet_prefix_name.db}_${var.azs[1]}"
+      new_bits = tonumber(local.subnet_cider_bits.db)
+    },
+    {
+      name     = "${local.subnet_prefix_name.db}_${var.azs[2]}"
+      new_bits = tonumber(local.subnet_cider_bits.db)
+    },
+    {
+      name     = "${local.subnet_prefix_name.public}_${var.azs[0]}"
+      new_bits = tonumber(local.subnet_cider_bits.public)
+    },
+    {
+      name     = "${local.subnet_prefix_name.public}_${var.azs[1]}"
+      new_bits = tonumber(local.subnet_cider_bits.public)
+    },
+    {
+      name     = "${local.subnet_prefix_name.public}_${var.azs[2]}"
+      new_bits = tonumber(local.subnet_cider_bits.public)
+    }
+  ]
+}
+
+locals {
+  subnets = {
+    public_subnet_cider = [for k, v in module.public_subnet_cider_blocks.network_cidr_blocks : v if can(regex("^${local.subnet_prefix_name.public}.*", k))]
+    public_subnet_name  = [for k, v in module.public_subnet_cider_blocks.network_cidr_blocks : k if can(regex("^${local.subnet_prefix_name.public}.*", k))]
+
+    db_subnet_cider = [for k, v in module.public_subnet_cider_blocks.network_cidr_blocks : v if can(regex("^${local.subnet_prefix_name.db}.*", k))]
+    db_subnet_name  = [for k, v in module.public_subnet_cider_blocks.network_cidr_blocks : k if can(regex("^${local.subnet_prefix_name.db}.*", k))]
+
+    private_subnet_cider = [for k, v in module.public_subnet_cider_blocks.network_cidr_blocks : v if can(regex("^${local.subnet_prefix_name.private}.*", k))]
+    private_subnet_name  = [for k, v in module.public_subnet_cider_blocks.network_cidr_blocks : k if can(regex("^${local.subnet_prefix_name.private}.*", k))]
+  }
+}
+
+
+
+resource "aws_eip" "nat_gateway" {
+  domain = "vpc"
+  tags = {
+    Name = "EIP_vpc_nat_gateway_igork_01"
+  }
+}
+
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "VPC_${var.owner_code_name}_${var.vpc_version}"
+  cidr = var.vpc_cidr_block
+
+  enable_nat_gateway = true
+  single_nat_gateway = true
+  reuse_nat_ips = true
+  external_nat_ip_ids = aws_eip.nat_gateway.*.id
+
+  create_database_subnet_group = false
+
+  azs                   = var.azs
```

Comment:
As we said we will remove this variable, please use "data aws_availability_zones" resource to find out which zones exist in region.